### ISSUE TITLE
Array based indexing with repeated indices.

### ISSUE THREAD

**Issue by miguelteixeiragomes** [arrayfire/arrayfire#1752]
When indexing an array with other arrays I get a strange behavior when the index array has repeated indices:

float idx_h[3] = { 3, 4, 3 };
array idx = array(3, idx_h);
array A = constant(0.0, 10);
A(idx) += 1.0
// I get the result { 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0 }

Is this undefined behavior? What can i expect when repeating indices.

**Comment by pavanky** [comment]
Arrayfire does not detect repeated indices. That said, the values at location 3 and 4 should both be non zero. I am not sure why only position 4 is 1.

@9prady9 can you check this ?

**Comment by miguelteixeiragomes** [comment]
Sorry, you are right, just edited the comment. So i shouldn't expect any sort of undefined behavior when repeating indices. Is there any away within ArrayFire to something similar and have the repeated indices count? I'm currently doing it by writing a CUDA kernel using atomics.

**Comment by pavanky** [comment]
@kaatish Is there an easy way to do a segmented reduction ?

**Comment by 9prady9** [comment]
@miguelteixeiragomes @pavanky In zero based index, the correct places {3, 4} are set to 1.0 in the output. It is not incorrect output. 

Note: If different operations are carried on different locations, then the output will be undefined because we can't guarantee which operation's result will end up in the output location.

**Comment by 9prady9** [comment]
I don't think this (automatic accumulation of results for same indices) can be done via indexing alone. Theoretically it can be done using atomic updates to identical locations, but I think that would cause a severe performance downgrade.

If accumulation of results for identical indices is the expected behavior, then some kind of sorting the indices+reduction has to be carried out.
